admin/class-yoast-dashboard-widget.php

```diff
@@ -134,6 +134,9 @@ public function localize_dashboard_script() {
 			'ryte_analyze'     => __( 'Analyze entire site', 'wordpress-seo' ),
 			'ryte_fetch_url'   => esc_attr( add_query_arg( 'wpseo-redo-onpage', '1' ) ) . '#wpseo-dashboard-overview',
 			'ryte_landing_url' => WPSEO_Shortlinker::get( 'https://yoa.st/rytelp' ),
+			'wp_version'       => substr ( $GLOBALS['wp_version'], 0, 3 ),
```

Comment:
This isn't 100% secure, because another plugin could overwrite `$GLOBALS`. A solution could be to cast `substr ( $GLOBALS['wp_version'], 0, 3 ),` to a float? As in: `(float) substr ( $GLOBALS['wp_version'], 0, 3 ),`.

Reply:
@johannadevos can you explain how casting `$GLOBALS['wp_version']` to a float mitigates the problem of that global potentially getting overwritten? `substr` already automatically casts the value to a string I think. Would it be better to access the global through `global $wp_version`?

Reply:
As I understand it, you couldn't prevent that global from being overwritten if someone has bad intentions, but you would mitigate the influence which that hack could have on your website (this is something @igorschoester explained to me), because floats are inherently safer due to being more constrained.

Reply:
Ah, yes, good point. I would think the `substr` kind off already protects against this since someone who overwrites the global could only inject a 3 character query parameter to the request, which seems practically useless. Do you think there is still merit in casting when taking this into account? Ie. is this a best practice that we apply everywhere?

Reply:
That's a good point too, it seems pretty safe then, but I think @jrfnl is the person to ask.

Reply:
@johannadevos Thanks for pinging me.

@omarreiss Am I correct that you just want the minor, not the patch version ?

In that case, for PHP I'd suggest using: `PHP_MAJOR_VERSION . '.' . PHP_MINOR_VERSION` which should be stable no matter what.

For the WP version:
* Using `global $wp_version` vs `$GLOBALS['wp_version']` makes no difference at all.
* @johannadevos's point about floats being safer than strings is valid, but @omarreiss' point about no three letter hacks being known, is valid too. Up to a point that is - what about things like `\0` (null bytes) and such ?

If you truly want to be on the safe side, I'd suggest not passing these variable to JS directly, but preparing the URL in PHP and passing that to JS instead.

Think along the lines of:
```php
// Before the array declaration:
$wp_minor           = substr( $GLOBALS['wp_version'], 0, 3 );
$has_classic_editor = ( is_plugin_active( 'classic-editor/classic-editor.php' ) ? '1' : '0' );
$post_feed_URL      = sprintf(
    'https://yoast.com/feed/widget/?wp_version=%.2f-%d&php_version=%d.%d',
    $wp_minor,
    $has_classic_editor,
    PHP_MAJOR_VERSION,
    PHP_MINOR_VERSION
);

// And then in the localization array:
    'post_feed'      => $post_feed_URL,
```

I also wonder if the resulting URL might need escaping if it is going to be used in HTML, but I don't know the full context of the PR.

Does this help ?

Reply:
Thanks - good to know!

Reply:
@jrfnl awesome input. I've followed the PHP major_minor version suggestion. I've checked but this data isn't output anywhere on the page except within a CDATA section where the entire blob is already escaped to prevent breaking out of the CDATA section. Resolving the conversation.

Reply:
+Thanks for raising this @johannadevos 😄 Learned a thing or two!

Reply:
@jrfnl @omarreiss Travis throws 2 errors, caused by the constants `PHP_MAJOR_VERSION` and `PHP_MINOR_VERSION`. Because they are not present in PHP 5.2.6<. Is this a problem which should be fixed, or can I ignore this?

Reply:
That should be fixed, but not in this PR. If everything else is ok you can merge it :)

Reply:
FYI: It's only relevant for the 0.001% of people still running WP on PHP 5.2.4/5.2.5 and can be easily fixed by (temporary - until November) adding the below to the `wpseo-main.php` file (or similar):

```php
if ( ! defined( 'PHP_MAJOR_VERSION' ) ) {
    define( 'PHP_MAJOR_VERSION', substr( PHP_VERSION, 0, 1 ) );
}
if ( ! defined( 'PHP_MINOR_VERSION' ) ) {
    define( 'PHP_MINOR_VERSION', substr( PHP_VERSION, 2, 1 ) );
}
```